drivers/spi/spi_npcx_spip.c

```diff
@@ -0,0 +1,436 @@
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_npcx_spip);
+
+#include "spi_context.h"
+
+/* Transfer this NOP value when tx buf is null */
+#define SPI_NPCX_SPIP_TX_NOP 0x00
+
+struct spi_npcx_spip_data {
+	struct spi_context ctx;
+	uint32_t src_clock_freq;
+	uint8_t bytes_per_frame;
+};
+
+struct spi_npcx_spip_cfg {
+	struct spip_reg *reg_base;
+	struct npcx_clk_cfg clk_cfg;
+#ifdef CONFIG_SPI_NPCX_SPIP_INTERRUPT
+	/* routine for configuring SPIP ISR */
+	void (*irq_cfg_func)(const struct device *dev);
+#endif
+	const struct pinctrl_dev_config *pcfg;
+};
+
+static int spi_npcx_spip_configure(const struct device *dev, const struct spi_config *spi_cfg)
+{
+	uint8_t prescaler_dividor;
+	const struct spi_npcx_spip_cfg *const config = dev->config;
+	struct spi_npcx_spip_data *const data = dev->data;
+	struct spip_reg *const reg_base = config->reg_base;
+	spi_operation_t operation = spi_cfg->operation;
+	uint8_t frame_size;
+
+	if (spi_context_configured(&data->ctx, spi_cfg)) {
+		/* This configuration is already in use */
+		return 0;
+	}
+
+	if (operation & SPI_HALF_DUPLEX) {
+		LOG_ERR("Half duplex mode is not supported");
+		return -ENOTSUP;
+	}
+
+	if (SPI_OP_MODE_GET(operation) != SPI_OP_MODE_MASTER) {
+		LOG_ERR("Only SPI controller mode is supported");
+		return -ENOTSUP;
+	}
+
+	if (operation & SPI_MODE_LOOP) {
+		LOG_ERR("Loopback mode is not supported");
+		return -ENOTSUP;
+	}
+
+	/*
+	 * If the GPIO CS configuration is not present, return error because the hardware CS is
+	 * not supported.
+	 */
+	if (!spi_cs_is_gpio(spi_cfg)) {
+		LOG_ERR("Only GPIO CS is supported");
+		return -ENOTSUP;
+	}
+
+	/* Get the frame length */
+	frame_size = SPI_WORD_SIZE_GET(operation);
+	if (frame_size == 8) {
+		data->bytes_per_frame = 1;
+		reg_base->SPIP_CTL1 &= ~BIT(NPCX_SPIP_CTL1_MOD);
+	} else if (frame_size == 16) {
+		reg_base->SPIP_CTL1 |= BIT(NPCX_SPIP_CTL1_MOD);
+		data->bytes_per_frame = 2;
+	} else {
+		LOG_ERR("Only support word sizes either 8 or 16 bits");
+		return -ENOTSUP;
+	}
+
+	if (IS_ENABLED(CONFIG_SPI_EXTENDED_MODES) &&
+	    (operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		LOG_ERR("Only single line mode is supported");
+		return -ENOTSUP;
+	}
+
+	/* Set the endianness */
+	if (operation & SPI_TRANSFER_LSB) {
+		LOG_ERR("Shift out with LSB is not supported");
+		return -ENOTSUP;
+	}
+
+	/*
+	 * Set CPOL and CPHA.
+	 * The following is how to map npcx spip control register to CPOL and CPHA
+	 *   CPOL    CPHA  |  SCIDL    SCM
+	 *   -----------------------------
+	 *    0       0    |    0       0
+	 *    0       1    |    0       1
+	 *    1       0    |    1       1
+	 *    1       1    |    1       0
+	 */
+	if (operation & SPI_MODE_CPOL) {
+		reg_base->SPIP_CTL1 |= BIT(NPCX_SPIP_CTL1_SCIDL);
+	} else {
+		reg_base->SPIP_CTL1 &= ~BIT(NPCX_SPIP_CTL1_SCIDL);
+	}
+	if ((operation & SPI_MODE_CPOL) ^ (operation & SPI_MODE_CPHA)) {
```

Comment:
I don't think the xor works here since CPOL and CPHA are two different bits in the mask, I think you want something like

`if (((operation & SPI_MODE_CPOL) == SPI_MODE_CPOL) != ((operation & SPI_MODE_CPHA) == SPI_MODE_CPHA) {`